src/Compilers/CSharp/Portable/Symbols/SubstitutedNamedTypeSymbol.cs

```diff
@@ -164,7 +164,17 @@ public sealed override IEnumerable<string> MemberNames
         {
             get
             {
-                return _unbound ? new List<string>(GetTypeMembersUnordered().Select(s => s.Name).Distinct()) : OriginalDefinition.MemberNames;
+                if (_unbound)
+                {
+                    return new List<string>(GetTypeMembersUnordered().Select(s => s.Name).Distinct());
+                }
+
+                if (IsTupleType)
```

Comment:
>IsTupleType [](start = 20, length = 11)

You only need to do this if it is a tuple type *and* has a set of tuple names different from the original definition's tuple names. I don't know, does that seem fragile?

Reply:
I like that the current logic is straightforward and I'm unsure that the potential saving/optimization warrant this extra subtlety ;)
So I think it's okay to leave as-is.

---
In reply to: [356906003](https://github.com/dotnet/roslyn/pull/39370#discussion_r356906003) [](ancestors = 356906003)